operators/src/util/raster_stream_to_geotiff.rs

```diff
@@ -205,6 +218,16 @@ fn gdal_writer<T: Pixel + GdalType>(
         band.write(window, window_size, &buffer)?;
     }
 
+    if as_cog {
+        // Override file with COG driver.
+
+        // Since COGs are written "overviews first, data second", we cannot generate a GeoTiff (where we first have to
+        // write the data in order to generate overviews) that fulfills this property. So, we have to do it as a
+        // separate step.
+
+        dataset.create_copy(&Driver::get("COG")?, output_file_path, &options)?;
```

Comment:
i don't think it is a good idea to write in the same file. what if the original file is not entirely in memory? then the file you read is overwriten while you still need data.

Reply:
2ad0dc1